src/deleteqso.c

```diff
@@ -47,8 +57,95 @@ void delete_qso(void)
+	    if (statbuf.st_size >= LOGLINELEN) {
+	        if (qtcdirection > 0) {
+		    // seek to last line to read it
+		    lseek(lfile, ((int)statbuf.st_size - LOGLINELEN), SEEK_SET);
+		    rc = read(lfile, logline, LOGLINELEN-1);
+		    // catch the band and mode (for QTC)
+		    strncpy(bandmode, logline, 5);
+		    bandmode[5] = '\0';
+		    call[14] = '\0';
+		    // catch the last callsign
+		    strncpy(call, logline+29, 14);
+		    i = strlen(call);
+		    // strip it
+		    for(i=strlen(call)-1; call[i] == ' '; i--);
+		    call[i+1] = '\0';
+		    // if qtc had been set up
+		    if (qtcdirection & 1) {
+			if ((qtcfile = open(QTC_RECV_LOG, O_RDWR)) < 0) {
+			    mvprintw(5, 0, "Error opening QTC received logfile.\n");
+			    sleep(1);
+			}
+			fstat(qtcfile, &qstatbuf);
+			if ((int)qstatbuf.st_size > QTCRECVCALLPOS) {
+			    look = 1;
+			    qtclen = 0;
+			    // iterate till the current line from back of logfile
+			    // callsigns is the current callsign
+			    // this works only for fixed length qtc line!
+			    lseek(qtcfile, 0, SEEK_SET);
+			    while (look == 1) {
+				lseek(qtcfile, ((int)qstatbuf.st_size - (91+qtclen)), SEEK_SET);
+				rc = read(qtcfile, logline, 90);
+                                logline[90] = '\0';
+
+				if (! (strncmp(call, logline+QTCRECVCALLPOS, strlen(call)) == 0 && strncmp(bandmode, logline, 5) == 0)) {
+				    look = 0;
+				}
+				else {
+				    qtclen += 91;
+				    qtc_dec(call, RECV);
+				}
+			    }
+			    rc = ftruncate(qtcfile, qstatbuf.st_size - qtclen);
+			    fsync(qtcfile);
+			}
+			close(qtcfile);
+		    }
+                    if (qtcdirection & 2) {
+                        if ((qtcfile = open(QTC_SENT_LOG, O_RDWR)) < 0) {
+                            mvprintw(5, 0, "Error opening QTC sent logfile.\n");
+                            sleep(1);
+                        }
+                        fstat(qtcfile, &qstatbuf);
+                        if ((int)qstatbuf.st_size > QTCSENTCALLPOS) {
+                            look = 1;
+                            qtclen = 0;
+			    s = nr_qsos;
+			    while(s >= 0 && qsoflags_for_qtc[s] != 1) {
+				s--;
+			    }
+                            // iterate till the current line from back of logfile
+                            // callsigns is the current callsign
+                            // this works only for fixed length qtc line!
+                            while (look == 1) {
+                                lseek(qtcfile, ((int)qstatbuf.st_size - (96+qtclen)), SEEK_SET);
+                                rc = read(qtcfile, logline, 95);
+                                if (! (strncmp(call, logline+QTCSENTCALLPOS, strlen(call)) == 0 && strncmp(bandmode, logline, 5) == 0)) {
+                                    look = 0;
+                                }
+                                else {
+                                    qtclen += 96;
+				    qtc_dec(call, SEND);
+				    qsoflags_for_qtc[s] = 0;
```

Comment:
Yes it is checked from left to right.
 But 's>=0'  is even true if s == 0 and then the 's--' is executed a last time resulting in s = -1. 
Or am I wrong?